### ISSUE TITLE
Build TreeFrog 1.24.0 for MSVC 2015 x86 configuration

### ISSUE THREAD

**Issue by amiart** [treefrogframework/treefrog-framework#248]
I see these lines in configure.bat:
```
if not "%MSCOMPILER%" == "" (
  devenv lz4\visual\VS2017\lz4.sln /project liblz4 /rebuild "Release|x64"
) else (
   // ...
)
```
and in corelib.pro:
```
  win32-msvc* {
    LIBS += ../3rdparty/lz4/visual/VS2017/bin/x64_Release/liblz4_static.lib
  } else {
    LIBS += ../3rdparty/lz4/lib/release/liblz4.a
  }
```

How can I build LZ4 library and TreeFrog for MSVC 2015 x86 configuration ?
Thanks.

**Comment by amiart** [comment]
To build the library for MSVC 2015 x86 change configure bat:

```
if not "%MSCOMPILER%" == "" (
  devenv lz4\visual\VS2010\lz4.sln /project liblz4 /rebuild "Release|Win32"
) else (
   // ...
)
```

and corelib.pro:

```
  win32-msvc* {
    LIBS += ../3rdparty/lz4/visual/VS2010/bin/Win32_Release/liblz4_static.lib
  } else {
    LIBS += ../3rdparty/lz4/lib/release/liblz4.a
  }
```

The configure must be run from elevated command line prompt, otherwise the mklink will not create a symbolic link to lz4 directory.

It would be nice If the configure can detect the compiler version and set the appropriate values.

**Comment by amiart** [comment]
The LZ4 library must be build in the same debug/release configuration as TreeFrog library, otherwise 
you get this error:

```
LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts
  with use of other libs; use /NODEFAULTLIB:library
```

So, for debug build you must change configure bat:

```
if not "%MSCOMPILER%" == "" (
  devenv lz4\visual\VS2010\lz4.sln /project liblz4 /rebuild "Debug|Win32"
) else (
   // ...
)
```
and corelib.pro:

```
  win32-msvc* {
    LIBS += ../3rdparty/lz4/visual/VS2010/bin/Win32_Debug/liblz4_static.lib
  } else {
    LIBS += ../3rdparty/lz4/lib/release/liblz4.a
  }
```

For release build see my previous comment.

**Comment by treefrogframework** [comment]
Try in 8539f65cdb91187dc38bf7966af2e6d48d7abe71.

**Comment by amiart** [comment]
The fix doesn't help. I get linker error:

```
link /NOLOGO /DYNAMICBASE /NXCOMPAT /DEBUG /DLL /SUBSYSTEM:CONSOLE /VERSION:1.25 /MANIFEST:embed /OUT:debug\treefrogd1.dll @C:\Users\ADMINI~1\AppData\Local\Temp\nm2487.tmp
   Creating library debug\treefrogd1.lib and object debug\treefrogd1.exp
tglobal.obj : error LNK2019: unresolved external symbol _LZ4_decompress_safe referenced in function "class QByteArray __cdecl Tf::lz4Uncompress(char const *,int)" (?lz4Uncompress@Tf@@YA?AVQByteArray@@PBDH@Z)
tglobal.obj : error LNK2019: unresolved external symbol _LZ4_compressBound referenced in function "class QByteArray __cdecl Tf::lz4Compress(char const *,int,int)" (?lz4Compress@Tf@@YA?AVQByteArray@@PBDHH@Z)
tglobal.obj : error LNK2019: unresolved external symbol _LZ4_compress_fast referenced in function "public: void __thiscall <lambda_c53249f24d8ac54f12508565e137cd5c>::operator()(char const *,int,int,class QByteArray &)const " (??R<lambda_c53249f24d8ac54f12508565e137cd5c>@@QBEXPBDHHAAVQByteArray@@@Z)
..\3rdparty\lz4\visual\liblz4_static.lib : warning LNK4272: library machine type 'x64' conflicts with target machine type 'X86'
debug\treefrogd1.dll : fatal error LNK1120: 3 unresolved externals
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\link.EXE"' : return code '0x460'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\nmake.exe"' : return code '0x2'
Stop.
```

The configure doesn't detect the architecture and build mode - the LZ4 library is build for x64 Release configuration even if I call configure with --enable-debug option and MSVC x86 compiler.

**Comment by amiart** [comment]
Fixed in b5d32def96e6c750cf207992c34497f6f36db442.
Thanks.

**Comment by treefrogframework** [comment]
Thanks.
